packages/job-worker/src/playout/adlib.ts

```diff
@@ -583,13 +586,17 @@ export function innerStartAdLibPiece(
 	cache: CacheForPlayout,
 	_rundown: DBRundown,
 	existingPartInstance: DBPartInstance,
-	newPieceInstance: PieceInstance
+	newPieceInstance: PieceInstance,
+	intoNextPart: boolean
```

Comment:
This is a boolean flag. Boolean flags are a bad practice since they make it crystal clear that the method now have atleast two responsibilities.

Reply:
Then what are those two responsibilities in this case?
Within this function or functions called by it, there's no significant logic that would depend on this argument. It is only used to set a certain property of the object, like other arguments.

Reply:
From the naming of the variable (which I assumed is correctly named) the one responsibility is for inserting into the current part, the other responsibility is for inserting into the next part.

> there's no significant logic that would depend on this argument

Then why is it even there if it's not something that is relevant for the function that it is being parsed to?

For the sake of completion (and because I wrote the snippet before I fully understood your question...) here is a pseudo example of one way to avoid a boolean flag
```
function someFunctionToStartTheFlow(piece: PieceInstance, ...) {
  const shouldBeInsertedIntoNextPart: boolean = // Logic to determine whether to insert into next part or not

  // Ignore that this is techincally not a valid ternary. Used for simplicity
  shouldBeInsertedIntoNextPart ? insertIntoNextPart(piece, ...) : insertIntoCurrentPart(piece, ...)
}

function insertIntoCurrentPart(piece: PieceInstance, ...) {
  const currentPart: PartInstance = // Logic to find the current part
  insertIntoPart(currentPart, piece)
}

function insertIntoNextPart(piece: PieceInstance, ...) {
  const nextPart: PartInstance = // Logic to find the next part
  insertIntoPart(nextPart, piece)
}

function insertIntoPart(part: PartInstance, piece: PienceInstance, ...) {
  // Logic to insert a piece into a part
}
```

Reply:
But I see you found a solution for this :)